libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -825,6 +919,23 @@ void AP_Baro::update(void)
             }
             float altitude = sensors[i].altitude;
             float corrected_pressure = sensors[i].pressure + sensors[i].p_correction;
+            // correct for ststic pressure position errors
+            Vector3f airspeed_vec_bf;
+            if (AP::ahrs().airspeed_vector_true(airspeed_vec_bf)) {
+                float pressure_error = 0.0f;
+                const float density = 1.225f * get_air_density_ratio();
```

Comment:
should use SSL_AIR_DENSITY

Reply:
we should call the compensation function only for sensors[i].type == BARO_TYPE_AIR